### PR TITLE
[14.0][FIX] base: fix `same_vat_partner_id` multicompany computation.

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
Description of the issue/feature this PR addresses:
Creating a new partner with no company selected should raise the same VAT partner warning if there is any other partner **in any** company with the same VAT.

Current behavior before PR:
1. Create a partner in company A with VAT "test123"
2. Click Save.
3. Create a new partner with company field in blank and set the VAT field to "test123". No duplicated VAT warning appears.

Desired behavior after PR is merged:
1. Create a partner in company A with VAT "test123"
2. Click Save.
3. Create a new partner with company field in blank and set the VAT field to "test123". Duplicated VAT warning appears.

@ForgeFlow

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by LoisRForgeFlow** [comment]
Hi @xmo-odoo and @nle-odoo have you had the chance to look at this topic?

Let me know if there is anything I can do to help you review faster.

Regards,

**Review by nle-odoo — APPROVED** [review]
reading history of current version of _compute_same_vat_partner_id (mainly here: https://github.com/odoo/odoo/pull/53701), this change seems good to me.

This also make the warning more symmetrical:

- if I have partner `{company: None, vat: 'money'}` and I create partner `{company: 1, vat: 'money'}` it conflicts

- if I have partner `{company: 1, vat: 'money'}` and I create partner `{company: None, vat: 'money'}` it doesn't conflicts before the change (and conflicts with this change)

But a review from accounting team is still needed.

**Comment by LoisRForgeFlow** [comment]
@nle-odoo Is there anything I can do to speed the process? or should I just wait for the final review of the accounting team?

Thanks

**Comment by lk-eska** [comment]
When should we expect the merge?

**Comment by LoisRForgeFlow** [comment]
Hi @nle-odoo just a kind reminder on this one. It would be great to get it merged.

**Comment by LoisRForgeFlow** [comment]
Hi @william-andre @smetl

Perhaps you could help to get this finally merged?

Thanks!

**Review by william-andre — APPROVED** [review]
Sorry about the delay

@robodoo r+

**Pull request closed without merging**
